src/connectors/menu/__tests__/connectMenu-test.js

```diff
@@ -681,4 +707,93 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/menu/js/#co
       });
     });
   });
+
+  describe('dispose', () => {
+    it('removes hierarchical refinements', () => {
+      const widget = makeWidget({
```

Comment:
Until we have a proper setup for integration tests we could focus only on `dispose`. Here we are lost inside the setup boilerplate. We can craft a `SearchParameters` by hand, it's not likely to change anytime soon.